Update "Chapter 2: Using Pulsar" documentation

Daeraxa opened the pull request:
Updating the "Using Pulsar" docs to remove Atom references etc and modify where necessary.

All the Atom Ch2 pages have now been added and should be working for our framework.
That means:
- Most `Atom` references rebranded as `Pulsar`
- Links to the `atom` repo have been replaced with `pulsar-edit` repo
- Some examples changes made such as the ppm search section for the packages it returns
- Links all changed to internal header links that work over the whole single index.md file.
- Pictures left untouched, we still need to have a single "theme" for redoing all our pictures of the application with atom references removed.
- Some references to `apm` and `ATOM_HOME` have been left and will be updated when these are done in core
- Some links to other chapters will not work yet and may need refactoring when those docs are included.
- Emojis for Linux🐧 , macOS🍎 and Windows🪟 will be replaced with ~~OS correct FA6 icons once #62 is merged.~~ as per Discord poll these are now replaced with LNX/MAC/WIN.

## Other comments

- Wording is pretty much exactly as it was in the flight manual - this PR is not about rewording the entire documentation. This is to get these docs usable ASAP so if there are comments about wording and general refactoring then please create a PR or an issue once this has been merged.
- Comments about the "platform switching" are welcome. I've tried my best to strike a decent balance between the "tab" switcher for the different OSs (at the cost of duplicated info) which is clunky and disruptive if used too often and the "inline" key shortcuts/paths etc. (like `🐧/🪟<kbd>Ctrl+,</kbd> - 🍎<kbd>Cmd+,</kbd>`) which work better when occasionally referenced in larger sections and don't cause unnecessary duplication of text.
	 - These were originally handled in the flight manual by a global platform switcher and html classes which made the markdown docs messy and hard to read, we don't have a direct equivalent of this in the new docs.

This has all been tested on the main branch of this repo and works just fine in `dev` from what I can see but please try the links etc. in case I have missed anything or if there are any missing images due to bad references etc.

kaosine commented on docs/docs/launch-manual/sections/using-pulsar/sections/atom-packages.md:
```diff
@@ -0,0 +1,133 @@
+### Pulsar Packages
```
I'd rename this file. It's going to be confusing that it's named atom-packages even if it's just as a include.

Daeraxa replied:
Its just a draft to get it shown as being worked on, not even nearly finished yet.

Daeraxa commented:
@softcode589 I don't think you should approve draft PRs, this is only a placeholder at the moment to show that I'm working on this chapter and there are a hell of a lot of mistakes and problems still in the content. Most of the links don't work and I've not even run it on dev.

Daeraxa commented:
The rest has been pushed, see the PR itself for the exact comments on this.

Daeraxa commented:
As per the discord poll I've removed the emojis and replaced with three letter abbreviations (LNX/MAC/WIN).
Package website URL has been added and TODO removed for it.

icecream17 commented on docs/docs/launch-manual/sections/using-pulsar/sections/editing-and-deleting-text.md:
```diff
@@ -0,0 +1,243 @@
+given maximum line length. You can format the current selection to have lines no
+longer than 80 (or whatever number `editor.preferredLineLength` is set to)
+characters using
+**_LNX/WIN_**: <kbd>Alt+Ctrl+Q</kbd> -
```
```suggestion
**_LNX/WIN_**: <kbd>Ctrl+Shift+Q</kbd> -
```

from https://github.com/pulsar-edit/pulsar-edit.github.io/pull/85

Daeraxa replied:
Thanks for the correction

icecream17 commented on docs/docs/launch-manual/sections/using-pulsar/sections/editing-and-deleting-text.md:
```diff
@@ -0,0 +1,243 @@
+**_LNX/WIN_**: <kbd>Ctrl</kbd> -
+**_MAC_**: <kbd>Cmd</kbd> key pressed down to select multiple regions of your
+text simultaneously.
+
```
```suggestion

Return to a single cursor with <kbd>Esc</kbd> or by clicking anywhere in the file to position a cursor there as normal.

```

https://github.com/atom/flight-manual.atom.io/pull/693/files
(btw, I can't access discord right now so if there's anything you want to tell me comment on github)

Daeraxa replied:
I think this is a nice clarification, thanks.

confused-Techie reviewed (dismissed):
This PR looks good.

It's been open for quite some time without to much activity. Unless there's any final suggestions or issues to raise, I'll go ahead and merge in a bit.

As always thanks for your contributions, this was a big one

icecream17 reviewed (dismissed):
the commands in basic customization are already mindblowing enough that I feel the mindblown emoji ctrl+left, right change if made would be a separate pr

confused-Techie commented:
I was going to initially merge this, but somehow missed this before.

The Test Deployment is failing on this PR. @Daeraxa would you be able to take a look at this?

Daeraxa commented:
There was a dodgy link that I somehow missed and also didn't see the failing test for some reason.
It should be fixed now.

confused-Techie reviewed (approved):
Thanks for fixing those issues. With my previous approval, and tests now passing lets merge!
